app/models/barbeque/job_queue.rb

```diff
@@ -16,4 +18,44 @@ class Barbeque::JobQueue < Barbeque::ApplicationRecord
   def sqs_queue_name
     SQS_NAME_PREFIX + name
   end
+
+  def update_policy!
+    client = Aws::SQS::Client.new
+    attrs = client.get_queue_attributes(
+      queue_url: queue_url,
+      attribute_names: ['QueueArn'],
+    )
+    queue_arn = attrs.attributes['QueueArn']
+    topic_arns = sns_subscriptions.map(&:topic_arn)
+    if topic_arns.present?
+      policy = generate_policy(queue_arn: queue_arn, topic_arns: topic_arns)
+    end
+
+    client.set_queue_attributes(
```

Comment:
Yes, it's what I intend.
If there're no topic_arns, the policy should be blank.